native/dlasq6.go

```diff
@@ -0,0 +1,107 @@
+// Copyright ©2015 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package native
+
+import "math"
+
+// Dlasq6 computes one dqd transform in ping-pong form with protection against
+// overflow and underflow. z has length at least 4*n and holds the qd array.
+// i0 is first index (should be zero-indexed rather than one indexed)
+// n0 is last index (should be zero-indexed)
+func (impl Implementation) Dlasq6(i0, n0 int, z []float64, pp int, dmin, dmin1, dmin2, dn, dnm1, dnm2 float64) (dminOut, dmin1Out, dmin2Out, dnOut, dnm1Out, dnm2Out float64) {
+	if len(z) < 4*(n0+1) {
+		panic(badZ)
+	}
+	if n0-i0-1 <= 0 {
+		return
+	}
+	safmin := dlamchS
+	j4 := 4*(i0+1) + pp - 4 // -4 rather than -3 for zero indexing
+	emin := z[j4+4]
+	d := z[j4]
+	dmin = d
+	if pp == 0 {
+		for j4loop := 4 * (i0 + 1); j4loop <= 4*((n0+1)-3); j4loop += 4 {
+			j4 := j4loop - 1 // Translate back to zero-indexed
```

Comment:
When we have more full integration tests we should attempt a cleanup of the code or switch it. This probably shouldn't be permanent.